packages/framework-provider-azure/src/library/events-adapter.ts

```diff
@@ -180,3 +180,30 @@ export async function storeSnapshot(
   logger.debug('Snapshot stored', snapshotEnvelope)
   return persistableEntitySnapshot
 }
+
+export async function storeDispatchedEvent(
+  cosmosDb: CosmosClient,
+  eventEnvelope: EventEnvelope,
+  config: BoosterConfig
+): Promise<boolean> {
+  const logger = getLogger(config, 'events-adapter#storeDispatchedEvent')
+  logger.debug('[EventsAdapter#storeDispatchedEvent] Storing EventEnvelope for event with ID: ', eventEnvelope.id)
+  try {
+    await cosmosDb
+      .database(config.resourceNames.applicationStack)
+      .container(config.resourceNames.dispatchedEventsStore)
+      .items.create({
+        eventId: eventEnvelope.id,
+      })
+    return true
+  } catch (e) {
+    if (e.code === 409) {
+      // If an item with the same ID already exists in the container, it will return a 409 status code.
+      // See https://learn.microsoft.com/en-us/rest/api/cosmos-db/http-status-codes-for-cosmosdb
+      logger.warn('[EventsAdapter#storeDispatchedEvent] Event has already been dispatched', eventEnvelope.id)
+    } else {
+      logger.error('[EventsAdapter#storeDispatchedEvent] Error storing dispatched event', e)
+    }
+  }
+  return false
```

Comment:
This will return **false** for any kind of error, which will prevent processing the event. I don't think that's a good idea. If there was a error storing the event in the dispatched table different than 409, we want to keep processing the event. It would be like a fallback to the actual behavior.

I would return false in case of a 409 and let any other error bubble up so that the core can decide what to do (which you already did in the method `EventStore.storeDispatchedEvent`